File: app/src/main/java/me/devsaki/hentoid/activities/sources/JjgirlsActivity.java
package me.devsaki.hentoid.activities.sources;

import me.devsaki.hentoid.enums.Site;

public class JjgirlsActivity extends BaseWebActivity {

    private static final String DOMAIN_FILTER = "jjgirls.com";
    private static final String[] DIRTY_ELEMENTS = {".unit-main",".unit-dt-blk",".unit-mobblk"};
    private static final String[] GALLERY_FILTER = {"jjgirls.com/.*/.*/.*/$"};

    Site getStartSite() {
        return Site.JJGIRLS;
    }

    @Override
    boolean allowMixedContent() {
        return false;
    }


    @Override
    protected CustomWebViewClient getWebClient() {
        addDirtyElements(DIRTY_ELEMENTS);
        CustomWebViewClient client = new CustomWebViewClient(GALLERY_FILTER, this);
        client.restrictTo(DOMAIN_FILTER);
        return client;
    }
}

File: app/src/main/java/me/devsaki/hentoid/activities/sources/JjgirlsActivity.kt
package me.devsaki.hentoid.activities.sources

import me.devsaki.hentoid.enums.Site

private const val DOMAIN_FILTER = "jjgirls.com"
private val GALLERY_FILTER = arrayOf("jjgirls.com/.*/.*/.*/$")
private val DIRTY_ELEMENTS = arrayOf(".unit-main", ".unit-dt-blk", ".unit-mobblk")

class JjgirlsActivityK : BaseBrowserActivity() {
    override fun getStartSite(): Site {
        return Site.JJGIRLS
    }

    override fun allowMixedContent(): Boolean {
        return false
    }


    override fun createWebClient(): CustomWebViewClient {
        val client = CustomWebViewClient(getStartSite(), GALLERY_FILTER, this)
        client.restrictTo(DOMAIN_FILTER)
        client.addRemovableElements(*DIRTY_ELEMENTS)
        return client
    }
}